django_tasks/backends/database/management/commands/db_worker.py

```diff
@@ -94,8 +101,9 @@ def start(self) -> None:
                 # If we're running in "batch" mode, terminate the loop (and thus the worker)
                 return None
 
-            # Wait before checking for another task
-            time.sleep(self.interval)
+            if self.running:
+                # Wait before checking for another task
+                time.sleep(self.interval)
```

Comment:
If ctrl-c has just interrupted a task, we should not sleep, but rather exit immediately.

Reply:
Make sense to me - nice catch! Could that be immortalised as a comment?

Reply:
I thought you might ask that :) I've added the comment.